src/Nethermind/Nethermind.Db/FullPruning/FullPruningDb.cs

```diff
@@ -187,6 +187,9 @@ private class PruningContext : IPruningContext
             public bool DuplicateReads { get; }
             private readonly FullPruningDb _db;
 
+            private long _counter = 0;
+            private ConcurrentBag<IBatch> _batches = new();
```

Comment:
Wouldn't ThreadStatic work better?

Reply:
I had VERY bad experience with ConcurrentBag in general, is it better here than ConcurrentQueue for example?

Reply:
Not sure how `ThreadStatic` would work here.... we would need to gather them back on commit.

Not sure if `ConcurrentQueue` have any benefit over this. I'm guessing, If I were to implement these container, there would be better optimization with `ConcurrentBag` due to lack of constraint on ordering.

Reply:
Really? That is unexpected. Ok, I'll swap it.

Reply:
`ConcurrentBag` is better if you have huge CPU contention (> 4M/sec) as it uses ThreadLocals and linked lists of items off them; so can actually add items in parallel; but it creates a linked list and allocates a node for each new add; whereas `ConcurrentQueue` reuses slots in an array